ci/deploy.sh

```diff
@@ -0,0 +1,96 @@
+
+# Set variables
+KETCH_TAG=""
+POOL=""
+INGRESS_TYPE=""
+INGRESS_ENDPOINT=""
+APP_NAME=""
+APP_ENV=""
+DOCKER_IMAGE=""
+REG_SECRET=""
+KETCH_YAML=""
+PROCFILE=""
+
+# if true, ketch will try to create app and pool for the deployment
+RESOURCE_CREATION=true
+
+# set colors for printing texts
+CLEAR='\033[0m'
+RED='\033[0;31m'
+
+# prints usage
+function usage() {
+  if [ -n "$1" ]; then
+    echo -e "${RED}👉 $1${CLEAR}\n";
+  fi
+
+  echo -e "Usage: $0 [-t --ketch-tag] [-o --pool] [-ig --ingress] [--endpoint] [-a --app] [-i --image] [-e --env] [-ig --ingress] [--registry-secret] [--ketch-yaml] [--procfile] [--skip-resource-creation]\n"
+  echo "  -t, --ketch-tag                 Ketch version. Default is latest."
+  echo "  -o, --pool                      Pool where your application should be deployed."
+  echo "  -a, --app                       App Name."
+  echo "  -e, --env                       Application environment variables."
+  echo "  -ig, --ingress                  Ingress type. Default is Traefik."
+  echo "  --endpoint                      Ingress IP address."
+  echo "  -i, --image                     The image that should be used with the application."
+  echo "  --registry-secret               A name of a Secret with docker credentials. This secret must be created in the same namespace of the pool."
+  echo "  --ketch-yaml                    The path to the ketch.yaml file."
+  echo "  --procfile	                  The path to Procfile. If not set, ketch will use the entrypoint and cmd from the image."
+  echo "  --skip-resource-creation        If set, ketch will NOT create app and pool for the deployment. Useful when resources already exist."
+  exit 1
+}
+
+# parse params and set variables with custom user inputs
+while [[ "$#" > 0 ]]; do case $1 in
+    -t|--ketch-tag) KETCH_TAG="$2"; shift;shift;;
+    -o|--pool) POOL="$2"; shift;shift;;
+    -ig|--ingress) INGRESS_TYPE="$2"; shift;shift;;
+    --endpoint) INGRESS_ENDPOINT="$2"; shift;shift;;
+    -a|--app) APP_NAME="$2"; shift;shift;;
+    -e|--env) APP_ENV="$2"; shift;shift;;
+    -i|--image) DOCKER_IMAGE="$2"; shift;shift;;
+    --registry-secret) REG_SECRET="$2"; shift;shift;;
+    --ketch-yaml) KETCH_YAML="$2"; shift;shift;;
+    --procfile) PROCFILE="$2"; shift;shift;;
+    --skip-resource-creation) RESOURCE_CREATION=false; shift;shift;;
+    *) usage "Unknown parameter passed: $1"; shift; shift;;
+esac; done
+
+
+# validate params
+if [ -z "$APP_NAME" ]; then usage "App Name required"; fi;
+if [ -z "$DOCKER_IMAGE" ]; then usage "Image for the app is required"; fi;
+
+# set defaults if not set by user
+if [ -z "$KETCH_TAG" ]; then 
+    KETCH_TAG=$(curl -s https://api.github.com/repos/shipa-corp/ketch/releases/latest | grep -Eo '"tag_name":.*[^\\]",' | head -n 1 | sed 's/[," ]//g' | cut -d ':' -f 2)
+fi
+
+# Install ketch binary at /usr/local/bin default location
+ curl -s https://raw.githubusercontent.com/shipa-corp/ketch/main/install.sh | TAG="${KETCH_TAG}" bash
+
```

Comment:
@vivek The ingress controller (Traefik or Istio), cluster issuer, and cert-manager can be pre-installed inside the cluster by the user. So, I skipped those installations as it can override old resources.

Please have a look at https://github.com/shipa-corp/ketch/pull/32/files#diff-794891fcb80950b0c81f330c416efdf5d35789fb56a91a18dcc25266d7cdccfdR5